cmd/goal/clerk.go

```diff
@@ -787,10 +787,17 @@ var signCmd = &cobra.Command{
 
 		for _, group := range groupsOrder {
 			txnGroup := []transactions.SignedTxn{}
-			for _, txn := range txnGroups[group] {
+			for idx, txn := range txnGroups[group] {
```

Comment:
doesn't `client.SignTransactionWithWalletAndSigner` already does that ? I think that this would only be needed in the case of `lsig.Logic != nil`.

Reply:
Sorry, do you mean move this if case into the lsig != nil check?

Reply:
I think so. But I could be wrong.
I think that the correct way to approach this is to have this tested.
We have some expect tests for the goal commands. I think that extending those to cover this use case wound't be a bad idea.

Reply:
Tweaked the logic but I'm struggling to find where the tests mentioned live

Reply:
The tests lives in /go-algorand/test/e2e-go/cli/goal/expect